src/foam/core/console/FileUpload.js

```diff
@@ -0,0 +1,1142 @@
+/**
+ * @license
+ * Copyright 2025 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.core.console',
+  name: 'FileUpload',
+  extends: 'foam.u2.Controller',
+
+  requires: [
+    'foam.dao.MDAO',
+    'foam.core.console.ColumnParser',
+    'foam.core.console.Mapping',
+    'foam.core.console.MappingsView',
+    'foam.core.console.PropertyChoiceView',
+    'foam.core.console.UploadService',
+    'foam.core.boot.CSpec',
+    'foam.core.fs.fileDropZone.FileDropZone',
+    'foam.core.fs.File',
+    'foam.u2.ActionView'
+  ],
+
+  imports: [ 'cSpecDAO' ],
+
+  mixins: [ 'foam.mlang.Expressions' ],
+
+  css: `
+    ^ {
+      padding: 20px;
+      max-width: 1200px;
+      margin: 0 auto;
+    }
+    ^ .step {
+      margin-bottom: 24px;
+      padding: 16px;
+      border: 1px solid #ddd;
+      border-radius: 8px;
+      background-color: #fafafa;
```

Comment:
Please use CSSTokens for all colors. You can just use the tokens provided in CSSTokens.js in case you dont want them to be component specific

Reply:
Anything that needs to be themed should use the primary color set, everything else can use grey and white and black